src/MoveUntilTouchTopicController.cpp

```diff
@@ -0,0 +1,216 @@
+#include <rewd_controllers/MoveUntilTouchTopicController.hpp>
+
+#include <functional>
+#include <pluginlib/class_list_macros.h>
+
+namespace rewd_controllers
+{
+//=============================================================================
+MoveUntilTouchTopicController::MoveUntilTouchTopicController()
+    : MultiInterfaceController{true}  // allow_optional_interfaces
+    , JointTrajectoryControllerBase{}
+    , mTaringCompleted{false}
+    , mForceThreshold{0.0}
+    , mTorqueThreshold{0.0}
+{
+}
+
+//=============================================================================
+MoveUntilTouchTopicController::~MoveUntilTouchTopicController() {}
+
+//=============================================================================
+bool MoveUntilTouchTopicController::init(hardware_interface::RobotHW* robot,
+                                    ros::NodeHandle& nh)
+{
+  // check that doubles are lock-free atomics
+  if (!mForceThreshold.is_lock_free()) {
+    ROS_ERROR(
+        "double atomics not lock-free on this system. Cannot guarantee "
+        "realtime safety.");
+    return false;
+  }
+
+  // load name of force/torque sensor handle from paramter
+  std::string ft_wrench_name;
+  if (!nh.getParam("forcetorque_wrench_name", ft_wrench_name)) {
+    ROS_ERROR("Failed to load 'forcetorque_wrench_name' parameter.");
+    return false;
+  }
+  // load name of force/torque tare handle from paramter
+  std::string ft_tare_name;
+  if (!nh.getParam("forcetorque_tare_name", ft_tare_name)) {
+    ROS_ERROR("Failed to load 'forcetorque_tare_name' parameter.");
+    return false;
+  }
+
+  // load force/torque saturation limits from parameter
+  if (!nh.getParam("sensor_force_limit", mForceLimit)) {
+    ROS_ERROR("Failed to load 'sensor_force_limit' paramter.");
+    return false;
+  }
+  if (!nh.getParam("sensor_torque_limit", mTorqueLimit)) {
+    ROS_ERROR("Failed to load 'sensor_torque_limit' paramter.");
+    return false;
+  }
+  if (mForceLimit < 0) {
+    ROS_ERROR("sensor_force_limit must be positive or zero");
+    return false;
+  }
+  if (mTorqueLimit < 0) {
+    ROS_ERROR("sensor_torque_limit must be positive or zero");
+    return false;
+  }
+
+  // subscribe to sensor data
+  forceTorqueDataSub = nh.subscribe(ft_wrench_name, 1000, &MoveUntilTouchTopicController::forceTorqueDataCallback, this);
```

Comment:
Do we need all 1000 messages in the queue? How frequently is this being called? I think the `RosJointStateClient` only  maintains a queue of size 1.

Reply:
My mistake.